### ISSUE TITLE
Is there a way to distinguish fenced code blocks from indented code blocks?

### ISSUE THREAD

**Issue by nsheff** [mkdocs/mkdocs#1720]
I have a markdown file that I am rendering with mkdocs that uses fenced code blocks and indented code blocks. In the mkdocs output, these code blocks are both rendered the same way, with the same style classes. Thus, it is impossible to distinguish them.

This arises because I want to include some jupyter notebooks in my mkdocs page. I am using `jupyter nbconvert` to convert the notebooks to `markdown`, and then rendering them with `mkdocs`. This works fine, but `jupyter nbconvert` uses fenced code blocks for input code, and indented code blocks for output code. 

This means that in a mkdocs render of a jupyter notebook, it is impossible to distinguish input from output code.

For example, see this gist:

https://gist.github.com/nsheff/4cc1ce7cc835e1f3fa2746de3f701e32

Rendered: http://code.databio.org/caravel/code_blocks/

**Comment by waylan** [comment]
MkDocs uses [Python-Markdown][1] to handle the Markdown parsing. Any matters pertaining to the behavior of the Markdown parser should be directed upstream to that project. 

But to answer your question, no there is no way to distinguish the two except that fenced code blocks can have a [class assigned to them][2], which is intended for use by JS based syntax highlighters. I don't see why you couldn't use a class which some other unique characteristic.

[1]: https://github.com/Python-Markdown/markdown
[2]: https://python-markdown.github.io/extensions/fenced_code_blocks/#language

**Comment by nsheff** [comment]
For future searchers, I was able to hack this by using my own Python-Markdown template file, with this content:

    {% extends 'markdown.tpl' %}
	{% block stream %}
	```.output
	{{ output.text }}
	```
	{% endblock stream %}

This adds the `output` class to all output code blocks. Unfortunately, it's not possible to add an `input` class to the input blocks because the default template uses the class to assign language for those blocks, and Python-Markdown doesn't support multi-class fenced blocks. I've raised [an issue](https://github.com/Python-Markdown/markdown/issues/775) there, which would be a pretty easy change and make this possible.
